arangod/RocksDBEngine/RocksDBCollection.cpp

```diff
@@ -1576,7 +1576,12 @@ Result RocksDBCollection::insertDocument(arangodb::transaction::Methods* trx,
       key.ref(),
       rocksdb::Slice(doc.startAs<char>(), static_cast<size_t>(doc.byteSize())));
   if (!s.ok()) {
-    return res.reset(rocksutils::convertStatus(s, rocksutils::document));
+    res.reset(rocksutils::convertStatus(s, rocksutils::document));
+    res.withError([&doc](result::Error& err) {
+      err.appendErrorMessage("; key: ");
+      err.appendErrorMessage(doc.get(StaticStrings::KeyString).copyString());
```

Comment:
`copyString`  will throw in case of `doc.get(StaticStrings::KeyString)` will return non-string slice. And we will lose the original error.  I see that actually there 'should' be correct _key attribute, maybe we make this intent clear with assertion like
`TRI_ASSERT(doc.get(StaticStrings::KeyString).isString())`